File: src/Entity/Hotel.php
<?php

namespace App\Entity;

use App\Repository\HotelRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: HotelRepository::class)]
class Hotel
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $NameH = null;

    #[ORM\Column(length: 255)]
    private ?string $Location = null;

    #[ORM\Column]
    private ?int $Rating = null;

    #[ORM\Column]
    private ?int $NumH = null;

    #[ORM\Column(length: 255)]
    private ?string $Description = null;

    #[ORM\Column(length: 255)]
    private ?string $PhotoUrl = null;

    #[ORM\OneToMany(mappedBy: 'IdHotel', targetEntity: Reservation::class)]
    private Collection $reservation;

    public function __construct()
    {
        $this->reservation = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNameH(): ?string
    {
        return $this->NameH;
    }

    public function setNameH(string $NameH): static
    {
        $this->NameH = $NameH;

        return $this;
    }

    public function getLocation(): ?string
    {
        return $this->Location;
    }

    public function setLocation(string $Location): static
    {
        $this->Location = $Location;

        return $this;
    }

    public function getRating(): ?int
    {
        return $this->Rating;
    }

    public function setRating(int $Rating): static
    {
        $this->Rating = $Rating;

        return $this;
    }

    public function getNumH(): ?int
    {
        return $this->NumH;
    }

    public function setNumH(int $NumH): static
    {
        $this->NumH = $NumH;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): static
    {
        $this->Description = $Description;

        return $this;
    }

    public function getPhotoUrl(): ?string
    {
        return $this->PhotoUrl;
    }

    public function setPhotoUrl(string $PhotoUrl): static
    {
        $this->PhotoUrl = $PhotoUrl;

        return $this;
    }

    /**
     * @return Collection<int, Reservation>
     */
    public function getReservation(): Collection
    {
        return $this->reservation;
    }

    public function addReservation(Reservation $reservation): static
    {
        if (!$this->reservation->contains($reservation)) {
            $this->reservation->add($reservation);
            $reservation->setIdHotel($this);
        }

        return $this;
    }

    public function removeReservation(Reservation $reservation): static
    {
        if ($this->reservation->removeElement($reservation)) {
            // set the owning side to null (unless already changed)
            if ($reservation->getIdHotel() === $this) {
                $reservation->setIdHotel(null);
            }
        }

        return $this;
    }
}

File: src/main/java/entities/Hotel.java
package entities;

import java.util.List;

public class Hotel {
    private int id;
    private String name_h;
    private String location;
    private float rating;
    private int num_h;

    public Hotel(int id, String name_h, String location, float rating, int num_h, String description, String photo_url, float price, List<Reservation> reservations) {
        this.id = id;
        this.name_h = name_h;
        this.location = location;
        this.rating = rating;
        this.num_h = num_h;
        this.description = description;
        this.photo_url = photo_url;
        this.price = price;
        this.reservations = reservations;
    }

    private String description;
    private String photo_url;
    private float price;

    public List<Reservation> getReservations() {
        return reservations;
    }

    public void setReservations(List<Reservation> reservations) {
        this.reservations = reservations;
    }

    private List<Reservation> reservations;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName_h() {
        return name_h;
    }

    public void setName_h(String name_h) {
        this.name_h = name_h;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }

    public int getNum_h() {
        return num_h;
    }

    public void setNum_h(int num_h) {
        this.num_h = num_h;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPhoto_url() {
        return photo_url;
    }

    public void setPhoto_url(String photo_url) {
        this.photo_url = photo_url;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }
}
